src/DuckNet.cpp

```diff
@@ -1,14 +1,13 @@
 #include "include/DuckNet.h"
 
-DuckNet* DuckNet::instance = NULL;
+#include <Update.h>
 
-DuckNet::DuckNet() { duckRadio = DuckRadio::getInstance(); }
-DuckNet* DuckNet::getInstance() {
-  if (instance == NULL) {
-    instance = new DuckNet();
-  }
-  return instance;
-}
+#include "DuckLogger.h"
+#include "include/Duck.h"
+
+DuckNet::DuckNet(Duck* duckIn):
```

Comment:
DuckNet acts as a client to the abstract Duck class as per this [diagram](https://docs.google.com/document/d/1BU741URkjpafsmTP-SOfwgTclh6ChB6yp93ItYjT7SM/edit). Having Duck own DuckNet manages the lifetime of DuckNet, since DuckNet relies on having access to a valid instance of Duck.

An alternative would be to make both Duck and DuckNet have static lifetimes.

Reply:
I agree the DuckNet should only exist if it has an existing instance of Duck.